tests/test_asetek_pro.py

```diff
@@ -0,0 +1,66 @@
+import unittest
```

Comment:
Can you add more tests to confirm that the messages being sent to the device are actually the ones that are expected so that if there is a refactor we can ensure that the protocol implementation has not changed accidentally.

Reply:
I've added some more unit test

Reply:
It seems much better, thanks.

But I'm also going skip reviewing the tests for now.